### PR TITLE
[AArch64] Add assembly/disassembly for FMOP4A (widening, 4-way) instructions

### PR DESCRIPTION
**Pull request by momchil-velikov**
The new instructions are described in
https://developer.arm.com/documentation/ddi0602/2024-09/SME-Instructions

### CODE REVIEW

File: llvm/lib/Target/AArch64/SMEInstrFormats.td [hunk 1]
```diff
@@ -5126,3 +5126,38 @@ class sme2_luti4_vector_vg4_strided<bits<2> sz, bits<2> op, string mnemonic>
...
+  let Inst{31-21} = 0b10000000001;
+  let Inst{20} = M;
+  let Inst{19-17} = Zm;
+  let Inst{16-10} = 0b0000000;
+  let Inst{9} = N;
+  let Inst{8-6} = Zn;
+  let Inst{5-2} = 0b0000;
+  let Inst{1-0} = ZAda;
+
+  let Constraints = "$ZAda = $_ZAda";
```
Comment (Lukacma) [inline llvm/lib/Target/AArch64/SMEInstrFormats.td:5148]: we should add `let Uses = [FPMR, FPCR]` here as it is FP8 instruction

Comment (momchil-velikov) [reply llvm/lib/Target/AArch64/SMEInstrFormats.td:5148]: It's in `AArch64SMEInstrInfo.td`.

Comment (Lukacma) [reply llvm/lib/Target/AArch64/SMEInstrFormats.td:5148]: Ah sorry missed that!

**Review by Lukacma — APPROVED** [review]
LGTM

**Comment by momchil-velikov** [comment]
Rebase, no changes.

**Pull request merged**